src/db.rs

```diff
@@ -85,6 +141,83 @@ pub fn insert_connection(connection: Connection) -> Result<Uuid> {
     Ok(nwuuid)
 }
 
+pub fn insert_policy(policy: &policy::Policy) -> Result<i64> {
+    let mut dbconn = get_dbconn().unwrap();
+
+    let mysqlstr = "INSERT INTO policy (allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, 
+                    allowed_build_ids, create_date, valid)
+                    VALUES(?, ?, ?, ?, ?, NOW(), 1)".to_string();
+
+    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+    /* create JSON for vector struct member variables */
+    let adj = serde_json::to_string(&policy.allowed_digests)?;
+    let apj = serde_json::to_string(&policy.allowed_policies)?;
+    let abidj = serde_json::to_string(&policy.allowed_build_ids)?;
+
+    trnsx.exec_drop(
+        mysqlstr,
+        (
+            &adj,
+            &apj,
+            policy.min_fw_api_major,
+            policy.min_fw_api_minor,
+            &abidj,
+        ),
+    )?;
+
+    let lid = trnsx.query_first("SELECT LAST_INSERT_ID()")?.unwrap();
```

Comment:
is there any way we can get this from exec without having to do a second query?

Reply:
I am not aware of a way to do this in one call.  I will keep looking, but I have not found a good solution because we need for mysql to complete the insert for the current connection.

Reply:
I used updated this function to use a QueryResult and extracted the last id inserted from this struct, so we are back to a single sql call.